bootloader/grub.go

```diff
@@ -446,19 +447,39 @@ func staticCommandLineForGrubAssetEdition(asset string, edition uint) string {
 	return string(cmdline)
 }
 
-var (
-	grubRecoveryModeTrustedAssets = []string{
-		// recovery mode shim EFI binary
-		"EFI/boot/bootx64.efi",
-		// recovery mode grub EFI binary
-		"EFI/boot/grubx64.efi",
-	}
+// grubBootAssetPath contains the paths for assets in the boot chain.
+type grubBootAssetPath struct {
+	shimBinary string
+	grubBinary string
+}
 
-	grubRunModeTrustedAssets = []string{
-		// run mode grub EFI binary
-		"EFI/boot/grubx64.efi",
-	}
-)
+// grubBootAssetsForArch contains the paths for assets for different
+// architectures in a map
+var grubBootAssetsForArch = map[string]grubBootAssetPath{
+	"amd64": {
+		shimBinary: filepath.Join("EFI/boot/", "bootx64.efi"),
+		grubBinary: filepath.Join("EFI/boot/", "grubx64.efi")},
+	"arm64": {
+		shimBinary: filepath.Join("EFI/boot/", "bootaa64.efi"),
+		grubBinary: filepath.Join("EFI/boot/", "grubaa64.efi")},
+}
+
+func getGrubBootAssetsForArch() grubBootAssetPath {
+	return grubBootAssetsForArch[arch.DpkgArchitecture()]
```

Comment:
the tests will need changes I think, because they will fail if run on a mismatched architecture. You might need to mock arch.DpkgArchitecture() to test this behavior and work on different platforms.

Reply:
I've done some changes now so tests will pass on different arches, and also to make sure that different arches are tested even when running the tests in a single arch.